WooCommerce/Classes/ViewRelated/Orders/Order Creation/NewOrder.swift

```diff
@@ -66,16 +64,44 @@ extension NewOrderHostingController: UIAdaptivePresentationControllerDelegate {
     }
 
     func presentationControllerDidAttemptToDismiss(_ presentationController: UIPresentationController) {
-        UIAlertController.presentDiscardChangesActionSheet(viewController: self, onDiscard: { [weak self] in
-            self?.viewModel.discardOrder()
-            self?.dismiss(animated: true, completion: nil)
-        })
+        presentDiscardChangesActionSheet(onDiscard: discardOrderAndDismiss)
+    }
+}
+
+/// Private methods
+///
+private extension NewOrderHostingController {
+    func checkUnsavedChangesAndDismissIfPossible() {
+        guard !viewModel.hasChanges else {
+            presentDiscardChangesActionSheet(onDiscard: discardOrderAndDismiss)
+            return
+        }
+
+        self.dismiss(animated: true)
+    }
+
+    func presentDiscardChangesActionSheet(onDiscard: (() -> Void)?) {
+        UIAlertController.presentDiscardChangesActionSheet(viewController: self, onDiscard: onDiscard)
+    }
+
+    func discardOrderAndDismiss() {
+        viewModel.discardOrder()
+        dismiss(animated: true)
+    }
+
+    func discardOrderAndPop() {
+        viewModel.discardOrder()
+        navigationController?.popViewController(animated: true)
     }
 }
 
 /// View to create a new manual order
 ///
 struct NewOrder: View {
+    /// Set this closure with UIKit dismiss code. Needed because we need access to the UIHostingController `dismiss` method.
+    ///
+    var dismiss: (() -> Void) = {}
```

Comment:
Totally agree with you 🙂 In fact, I was thinking of changing the name to use something like `cancelButtonTapHandler`, but then, I thought that this closure is because this view is embedded in a `UIHostingController`, so I think we should still use `dismissHandler` to be clear what the closure should do. 
WDYT?

PS: Naming problems, I love them ❤️ (no irony 🙂)

Reply:
Yes, `dismissHandler` seems to be the best candidate in this case.